File: app/src/main/java/me/zhanghai/android/douya/network/api/info/AuthenticationResponse.java
/*
 * Copyright (c) 2017 Zhang Hai <Dreaming.in.Code.ZH@Gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.douya.network.api.info;

import com.google.gson.annotations.SerializedName;

public class AuthenticationResponse {

    @SerializedName("access_token")
    public String accessToken;

    @SerializedName("douban_user_name")
    public String userName;

    @SerializedName("douban_user_id")
    public long userId;

    @SerializedName("expires_in")
    public long accessTokenExpiresIn;

    @SerializedName("refresh_token")
    public String refreshToken;
}

File: app/src/main/java/me/zhanghai/android/douya/api/info/AuthenticationResponse.kt
/*
 * Copyright (c) 2019 Hai Zhang <dreaming.in.code.zh@gmail.com>
 * All Rights Reserved.
 */

package me.zhanghai.android.douya.api.info

import com.squareup.moshi.Json
import com.squareup.moshi.JsonClass

@JsonClass(generateAdapter = true)
data class AuthenticationResponse(
    @Json(name = "access_token")
    val accessToken: String,
    @Json(name = "douban_user_name")
    val userName: String,
    @Json(name = "douban_user_id")
    val userId: Long,
    @Json(name = "expires_in")
    val accessTokenExpiresIn: Long,
    @Json(name = "refresh_token")
    val refreshToken: String
)
